pkg/webhook/controlplane/ensurer.go

```diff
@@ -328,91 +259,37 @@ var (
 	}
 )
 
-func ensureKubeAPIServerVolumeMounts(c *corev1.Container, csiEnabled, csiMigrationComplete bool) {
-	if csiEnabled && csiMigrationComplete {
-		c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount.Name)
-		return
-	}
-
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount)
+func ensureKubeAPIServerVolumeMounts(c *corev1.Container) {
+	c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount.Name)
+}
 
-	// TODO: remove this in a future version.
-	// previous openstack versions (no CSI) ensure a volume mount with name `usr-share-ca-certificates` with path `/usr/share/ca-certificates`
-	// however Gardener version > 1.10.0 adds (if feature flag `MountHostCADirectories` is enabled) an API Server volume with name `usr-share-cacerts` also with path `/usr/share/ca-certificates`
-	// volume mount `usr-share-ca-certificates` needs to be removed to not have a duplicate `/usr/share/ca-certificates` mount
+func ensureKubeControllerManagerVolumeMounts(c *corev1.Container) {
+	c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount.Name)
+	c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, etcSSLVolumeMount.Name)
 	c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, usrShareCACertificatesVolumeMount.Name)
 }
 
-func ensureKubeControllerManagerVolumeMounts(c *corev1.Container, csiEnabled, csiMigrationComplete bool) {
-	if csiEnabled && csiMigrationComplete {
-		c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount.Name)
-		c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, etcSSLVolumeMount.Name)
-		c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, usrShareCACertificatesVolumeMount.Name)
-		return
-	}
-
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount)
-	// Host certificates are mounted to accommodate OpenStack endpoints that might be served with a certificate
-	// signed by a CA that is not globally trusted.
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, etcSSLVolumeMount)
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, usrShareCACertificatesVolumeMount)
+func ensureKubeAPIServerVolumes(ps *corev1.PodSpec) {
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderDiskConfigVolume.Name)
```

Comment:
Similar to https://github.com/gardener/gardener-extension-provider-gcp/pull/533#discussion_r1050822910